### PR TITLE
Small cpu_comparison "pythonic" improvements

### PR DESCRIPTION
**Pull request by makslevental**
I was gonna start working on https://github.com/nod-ai/iree-amd-aie/issues/581, which is downstream of one of the comparisons, and I started using the script and then got distracted.

### CODE REVIEW

File: build_tools/ci/cpu_comparison/run_test.py [hunk 1]
```diff
@@ -114,83 +81,69 @@ def generate_aie_output(
...
+        if config.verbose:
+            print(f"Running {i}/{config.num_repeats} compares")
+        shell_out(config.reset_npu_script)
+        start = time.monotonic_ns()
+        shell_out(run_args, config.output_dir, config.verbose)
+        run_time += time.monotonic_ns() - start
+
+    print(f"Time spent in compilation: {compile_time // 1e6} [ms]")
+    print(
+        f"Time spent in running the model: {run_time // config.num_repeats // 1e6} [ms]"
```
Comment (newling) [inline build_tools/ci/cpu_comparison/run_test.py:145]:
```suggestion
        f"Mean time over {config.num_repeats} run(s) of model: {run_time // config.num_repeats // 1e6} [ms]"
```

**Review by newling — CHANGES_REQUESTED** [review]
Thanks for the improvements. n_repeats vs num_repeats, not clear what's going on is num-repeats a global default? 

Please run black if you haven't already

**Comment by makslevental** [comment]
> Thanks for the improvements. n_repeats vs num_repeats, not clear what's going on is num-repeats a global default?

err I'm blind - I didn't notice `n_repeats`.

> Please run black if you haven't already

I have black bound to `ctrl+shift+b` and I run it compulsively after every edit lol

**Comment by makslevental** [comment]
> not clear what's going on is num-repeats a global default?

Since I didn't realize `n_repeats` existed, I patched in a different semantic for `num_repeat`s (just repeating the run). But actually the current/existing semantic you have for `n_repeats` is the right one for this test. So I'm gonna remove the changes associated with `num_repeats`.

**Comment by newling** [comment]
> > not clear what's going on is num-repeats a global default?
> 
> Since I didn't realize `n_repeats` existed, I patched in a different semantic for `num_repeat`s (repeating only the run). But actually the current/existing semantic you have for `n_repeats` is the right one for this test. So I'm gonna remove the changes associated with `num_repeats`.

I was wondering about this actually, repeating the compile does hurt test time quite badly and I'm not sure it's worth it. Ideally we could have the option for both, and if we're repeating the compile it'd be nice if we could actually diff the artifacts (xclbin?) to see if compilation is deterministic. xclbin might contain timestamps or something which this not possible though. 

TL;DR: my current thinking is that if we have to choose one or the other, we shouldn't be repeating the compilation, just the run... FFS.

**Review by newling — COMMENTED** [review]
n_repeats is still confusing: run_test takes as 2 arguments for the number of repeats, one via config (config.n_repeats) and one via n_repeats. 
 
In this PR, can you please remove all changes to n_repeats, and we can get back to it later? Everything else looks good to me

File: build_tools/ci/cpu_comparison/run_test.py [hunk 2]
```diff
@@ -449,33 +403,22 @@ def run_all(argv):
...
-    assert_num_args(argv)
-
-    output_dir = os.path.realpath(argv[0])
-    os.makedirs(output_dir, exist_ok=True)
-    validate_directory(output_dir)
-
-    iree_install_dir = os.path.realpath(argv[1])
-    validate_directory(iree_install_dir)
+    if output_dir.exists():
+        shutil.rmtree(output_dir)
```
Comment (newling) [inline build_tools/ci/cpu_comparison/run_test.py:407]: This is dangerous! What if the user has something in output_dir that they don't want removed, for example maybe they accidentally provide the path to their home directory for output_dir.

**Review by newling — CHANGES_REQUESTED** [review]

File: build_tools/ci/cpu_comparison/run_test.py (hunk 2 above)
Comment (makslevental) [reply build_tools/ci/cpu_comparison/run_test.py:407]: is it more or less dangerous than the issue you had a little while ago where tests were passing even though nothing was compiling? i'm doing this by hand right now (ie in an outer script). anyway i can take it out and leave it to the user to make the decision.

Comment (newling) [reply build_tools/ci/cpu_comparison/run_test.py:407]: Maybe add a script flag which by default is "do not remove"

Comment (makslevental) [reply build_tools/ci/cpu_comparison/run_test.py:407]: too many script flags turns into "programming by script flags". i'll just leave it out.

Comment (newling) [reply build_tools/ci/cpu_comparison/run_test.py:407]: Ok. Yeah I'm not sure what a good solution is, sorry. Maybe walk the dir and remove xclbins, that would avoid the issue we had before when compilation was failing (although is seems arbitrary and complex).

**Comment by makslevental** [comment]
@newling all set

**Review by newling — APPROVED** [review]
Thanks!
